### PR TITLE
Basic geocoding feature

### PR DESCRIPTION
**Pull request by VictorVelarde**
Geocode a text, using Data Services API and *geocodeStreetPoint* --> lib/sdk

This is the result:
image

You can test it at: https://cra-template-carto-git-feature-geocoding.carto-frontend.vercel.app/

IMPORTANT_ You will need to Login (OAuth) to get results, otherwise you will get a console error

### CODE REVIEW

**Comment by VictorVelarde** [comment]
I need to rebase (and clean) this branch before moving on, after the latest changes and refactor in develop.

**Comment by VictorVelarde** [comment]
I have already cleaned this branch, and point the PR to 'simple-oauth' branch, as that one is required before

**Comment by VictorVelarde** [comment]
I have just seen the other PR merged, so I'm gonna fix the conflict

File: template/src/lib/sdk/widgets/GeocoderWidget.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  input: {
+    marginLeft: theme.spacing(2),
+    flex: 1,
+  },
+}));
+
+export default function GeocoderWidget(props) {
+  const oauthCredentials = useSelector(selectOAuthCredentials);
+  const credentials = useSelector((state) => state.carto.credentials);
+  const getDataServicesCredentials = () => {
```
Comment (alasarr) [inline template/src/lib/sdk/widgets/GeocoderWidget.js:25]: How about this?

```javascript
const credentials = useSelector(selectOAuthCredentials) ||
                                  useSelector((state) => state.carto.credentials);
```
```

File: template/src/lib/sdk/widgets/GeocoderWidget.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  };
+
+  const handleInput = (e) => {
+    if (e.target.value === '') {
+      updateMarker(null);
+    }
+  };
+
+  const handleKeyPress = async (e) => {
+    const credentials = getDataServicesCredentials();
```
Comment (alasarr) [inline template/src/lib/sdk/widgets/GeocoderWidget.js:47]: We need a debounce here to save credits

File: template/src/lib/sdk/widgets/GeocoderWidget.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  };
+
+  const handleGeocodeError = (error) => {
+    dispatch(setError(`Geocoding error: ${error.message}`));
+  };
+
+  useEffect(() => {
+    // layer to display the geocoded direction marker
+    dispatch(
+      addLayer({
```
Comment (alasarr) [inline template/src/lib/sdk/widgets/GeocoderWidget.js:93]: We should have some mechanism to removeLayer

**Review by alasarr — CHANGES_REQUESTED** [review]

File: template/src/lib/sdk/widgets/GeocoderWidget.js (hunk 2 above)
Comment (VictorVelarde) [reply template/src/lib/sdk/widgets/GeocoderWidget.js:47]: The geocoding is just launched on 'Enter' click

**Pull request merged**